scripts/AbinsModules/LoadVASP.py

```diff
@@ -0,0 +1,328 @@
+    these calculations only sample the Gamma-point in reciprocal space.
+
+    """
+
+    def __init__(self, input_ab_initio_filename: str = None) -> None:
+        """
+
+        :param input_ab_initio_filename: name of file with phonon data (foo.phonon)
+        """
+        super().__init__(input_ab_initio_filename=input_ab_initio_filename)
+        self._ab_initio_program = "VASP"
+
+    def read_vibrational_or_phonon_data(self,
+                                        logger: Logger = None,
+                                        ) -> AbinsData:
+        input_filename = self._clerk.get_input_filename()
+
+        if not os.path.isfile(input_filename):
+            raise IOError("Could not find file: {}".format(input_filename))
+
+        if input_filename[-4:] == '.xml':
+            data = self._read_vasprun(input_filename)
+            self._num_atoms = len(data['atoms'])
+            self._num_k = 1
+
+        elif 'OUTCAR' in input_filename:
+            if logger is None:
+                import mantid.kernel
+                mantid_logger: mantid.kernel.Logger = mantid.kernel.logger
+                logger = mantid_logger
+            logger.warning("Reading OUTCAR file. This feature is included for development purposes "
+                           "and may be removed in future versions. Please use the vasprun.xml file where possible.")
+            data = self._read_outcar(input_filename)
+            self._num_atoms = len(data['atoms'])
+            self._num_k = 1
+
+        else:
+            raise ValueError('Cannot guess format from filename "{}". Expected'
+                             ' *.xml or *OUTCAR*'.format(input_filename))
+
+        self.save_ab_initio_data(data=data)
+        return self._rearrange_data(data=data)
+
+    @classmethod
+    def _read_outcar(cls, filename) -> Dict[str, Any]:
+        # Vibration calculations within Vasp only calculate the Hessian
+        # within the calculation cell: i.e. they only include Gamma-point
+
+        file_data = {'k_vectors': np.array([[0, 0, 0]], dtype=FLOAT_TYPE),
+                     'weights': np.array([1.], dtype=FLOAT_TYPE),
+                     'atoms': {}}
+
+        parser = AbinsModules.GeneralAbInitioParser()
+
+        with open(filename, 'rb') as fd:
+            # Lattice vectors are found first, with block formatted e.g.
+            #
+            #  Lattice vectors:
+
+            # A1 = (  17.7648860000,   0.0000000000,   0.0000000000)
+            # A2 = (   0.0000000000,  18.0379140000,   0.0000000000)
+            # A3 = (   0.0000000000,   0.0000000000,  18.3144580000)
+            unit_cell = np.empty((3, 3), dtype=float)
+            _ = parser.find_first(file_obj=fd, msg='Lattice vectors:')
+            _ = fd.readline() # skip a line
+
+            for i in range(3):
+                lattice_line = fd.readline().decode("utf-8")
+                float_re = r'\s+(\d+\.\d+)'
```

Comment:
Could we pull the various regex strings used in the file out to be constants at the top. I think this would simplify the reading of the code and also make potentially changing the regex expressions in the future easier.

Reply:
Have moved regexes to top of OUTCAR reading function with some examples of what they will be chewing on